### DISCUSSION TITLE
Sub application proxy docs showing same thing as main application

### DISCUSSION THREAD

**Discussion by felipeamaral** [fastapi/fastapi#8272] [Questions] [answered]
**Description**
Hey all,

I'm trying to use the sub applications proxy feature (https://fastapi.tiangolo.com/tutorial/sub-applications-proxy/) but it is acting quite weird. Even when i try to use the example, it doesn't work as expected. Is there any other configuration that I have to set? 

For example,
/docs works fine
/subapi/docs shows the same as /docs (it is using /openapi.json instead of /subapi/openapi.json, and they are different.)

Thanks

**Comment by tiangolo** [comment]
Make sure that the sub-application has the `openapi_prefix` parameter:

```Python
subapi = FastAPI(openapi_prefix="/subapi")
```

Did you try with the example files exactly as in the tutorial? If you use those, do they work as expected?

**Comment by felipeamaral** [comment]
I tried with my own application and with the example files (copy & paste) and it didn't work. Both using openapi_prefix="/...")

The weird thing is that /subapi/openapi.json works fine. But the /subapi/docs does the request to /openapi.json

**Comment by tiangolo** [comment]
@felipeamaral let's discard first with the same code as in the docs, if you have a single file, just like in the docs, with:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/app")
def read_main():
    return {"message": "Hello World from main app"}


subapi = FastAPI(openapi_prefix="/subapi")


@subapi.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API"}


app.mount("/subapi", subapi)
```

Does it work?

Do you see the different docs at `/docs` and `/subapi/docs`?

**Comment by felipeamaral** [comment]
@tiangolo it doesn't work, only shows `/docs`.

My guess is that` /subapi/docs` if getting the wrong openapi.json, look at the requests

![image](https://user-images.githubusercontent.com/1967055/56906130-ff656b80-6a77-11e9-8b87-59ab981bcee0.png)

Both `/docs` and` /subapi/docs` send the request for `/openapi.json.`

But If I access `/subapi/openapi.json` it shows subapi information.

**Comment by tiangolo** [comment]
That example in the comment above should work in the latest version (at least). I tested it locally before adding the comment, I just tested it again. And it's part of the tests. In this case, it might be related to your environment or something similar.

Which version of FastAPI do you have?

**Comment by felipeamaral** [comment]
Hey @tiangolo it was a version problem 

Thx for you help :+1:

**Reply by lyutian** [reply]
Anyone can tell which version will support the official example code exactly?

My 0.54.1 FastAPI need to manually add `openapi_prefix` parameter in `FastAPI()`, which makes me uncomfortable.
I don't want any related string (like `"/subapi"` here) between two independent applications.

**Comment by tiangolo** [comment]
Great! Thanks for reporting back and closing the issue.

**Comment by ilyasotkov** [comment]
Documentation link above is returning 404 now, the new link is

https://fastapi.tiangolo.com/advanced/sub-applications-proxy/

**Comment by YuriiMotov** [comment, marked as answer]
This works fine without `openapi_prefix` (it's deprecated) in current version (0.116.0):

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/app")
def read_main():
    return {"message": "Hello World from main app"}

subapi = FastAPI()

@subapi.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API"}

app.mount("/subapi", subapi)
```

![image](https://github.com/user-attachments/assets/9f666c44-1f9d-400d-a136-e8a4b6a87cef)


![image](https://github.com/user-attachments/assets/7ce9633d-5591-4075-8262-5fde5b44fdb2)
